tests/foreman/upgrade/conftest.py

```diff
@@ -0,0 +1,210 @@
+import datetime
+import functools
+import json
+import logging
+import os
+
+
+from pytest import fixture, hookspec
+
+from robottelo.decorators.func_locker import lock_function
+
+LOGGER = logging.getLogger('robottelo')
+
+pre_upgrade_failed_tests = []
+
+
+PRE_UPGRADE_TESTS_FILE_OPTION = 'pre_upgrade_tests_file'
+PRE_UPGRADE_TESTS_FILE_PATH = '/var/tmp/robottelo_pre_upgrade_failed_tests.json'
+PRE_UPGRADE = False
+POST_UPGRADE = False
+PRE_UPGRADE_MARK = 'pre_upgrade'
+POST_UPGRADE_MARK = 'post_upgrade'
+TEST_NODE_ID_NAME = '__pytest_node_id'
+
+__initiated = False
+
+
+def log(message, level="DEBUG"):
+    """Pytest has a limitation to use logging.logger from conftest.py
+    so we need to emulate the logger by std-out the output
+    """
+    now = datetime.datetime.now()
+    full_message = "{date} - conftest - {level} - {message}\n".format(
+        date=now.strftime("%Y-%m-%d %H:%M:%S"),
+        level=level,
+        message=message
+    )
+    print(full_message)  # noqa
+    with open('robottelo.log', 'a') as log_file:
+        log_file.write(full_message)
+
+
+# todo remove when upgrade_tests will be python 3 compatible
```

Comment:
@ldjebran , PRs are now merged to run upgrade scenarios on python3.

Reply:
@jyejare still not compatible as having error when importing
```
  from upgrade_tests.helpers.scenarios import create_dict, get_entity_data
E     File "/home/dlezz/.pyenv/versions/robottelo/lib/python3.6/site-packages/upgrade_tests/helpers/scenarios.py", line 164
E       print hammer.hammer('subscription upload --file {0} '


```

Reply:
Yeah @ldjebran , That is the scenario yet to be moved to py3. I think @vijay8451 is going to have a look into it else I will.